### PR TITLE
update azure deployment status

### PR DESCRIPTION
**Pull request by yaananth**

### CODE REVIEW

**Comment by pascalberger** [comment]
You wanted to add the updated resjson files with this PR?

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 [hunk 1]
```diff
@@ -95,14 +93,67 @@ if($WebSiteLocation)
...
+    $status = 3 #failed
+    if(!$publishAzureWebsiteError) {
+        $status = 4 #succeeded
+    }
+
+    $username = $azureWebSite.PublishingUsername
+    $securePwd = ConvertTo-SecureString $azureWebSite.PublishingPassword -AsPlainText -Force
+    $credential = New-Object System.Management.Automation.PSCredential ($username, $securePwd)
+    
+    $author = Get-TaskVariable $distributedTaskContext "build.sourceVersionAuthor"
```
Comment (ericsciple) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: For AT/DT mismatch, consider handling this gracefully to short-circuit if this is required.

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 [hunk 2]
```diff
@@ -95,14 +93,67 @@ if($WebSiteLocation)
...
+    $deploymentId = Get-TaskVariable $distributedTaskContext "build.sourceVersion" #let's use commitId as unique deploymentId
+    $message = Get-TaskVariable $distributedTaskContext "build.sourceVersionMessage"
+    $buildId = Get-TaskVariable $distributedTaskContext "build.buildId"
+    
+    $collectionUrl = "$env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI".TrimEnd('/')
+    $teamproject = "$env:SYSTEM_TEAMPROJECTID"
+    $buildUrl = [string]::Format("{0}/{1}/_build#buildId={2}&_a=summary",$collectionUrl,$teamproject, $buildId)
```
Comment (ericsciple) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: Consider simplifying string formatting using: "$collectionUrl/$teamproject/_build#buildId=$buildId&_a=summary"

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 [hunk 3]
```diff
@@ -95,14 +93,67 @@ if($WebSiteLocation)
...
+    
+    $body = "{
```
Comment (ericsciple) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: To handle json special chars:
ConvertTo-Json (New-Object -TypeName psobject -Property @{
  'status' = 'Some status'
  'message' = 'Some message'
})

**Comment by yaananth** [comment]
@pascalberger Those are generated files, someone didn't do it right before and hence they showed up on mine.

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 (hunk 2 above)
Comment (yaananth) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: I guess it is nice to be explicit, may be string format handles any weird characters ( not that I expect in this case anyways )

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 (hunk 3 above)
Comment (yaananth) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: will update

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 [hunk 4]
```diff
@@ -95,14 +93,67 @@ if($WebSiteLocation)
...
+    
+    $body = ConvertTo-Json (New-Object -TypeName psobject -Property @{
+        status = $status
+        message = $message
+        author = $author
+        deployer = 'VSTS'
+        details = $buildUrl       
+    })
+    
+    $url = [string]::Format("https://{0}.scm.azurewebsites.net/deployments/{1}",$WebSiteName,$deploymentId)
```
Comment (ericsciple) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: look at system.uri encode functions:
[system.uri] | gm -static

**Pull request merged**